### ISSUE TITLE
False Negative: SummaryJavadocCheck

### ISSUE THREAD

**Issue by Vyom-Yadav** [checkstyle/checkstyle#11407]
have read check documentation: https://checkstyle.sourceforge.io/config_javadoc.html#SummaryJavadoc
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java

/var/tmp $ cat config.xml
```
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="SummaryJavadoc"/>
    </module>
</module>
```
```
/var/tmp $ cat Test.java
```
```java
/**
 * This is 1 version
 */
public class Test {
}

/**
 * This is 1.0 version
 */
class Test2 {
}
```
```
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.0-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /var/tmp/Test.java:1: First sentence of Javadoc is missing an ending period. [SummaryJavadoc]
Audit done.
Checkstyle ends with 1 errors.
```
---

**Describe what you expect in detail.**

```
/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-10.0-all.jar -c config.xml Test.java
Starting audit...
[ERROR] /var/tmp/Test.java:1: First sentence of Javadoc is missing an ending period. [SummaryJavadoc]
[ERROR] /var/tmp/Test.java:7: First sentence of Javadoc is missing an ending period. [SummaryJavadoc]
Audit done.
Checkstyle ends with 2 errors.
```

The `.` in `1.0` is falsely considered as the ending period which causes this bug.

**Comment by Vyom-Yadav** [comment]
On it.

**Comment by nick-mancuso** [comment]
Please print out AST with javadoc for your example, and share it in a comment here.

**Comment by romani** [comment]
Please share javadoc tool output on this, how html look like for this methods?

**Comment by Vyom-Yadav** [comment]
AST:-
```
COMPILATION_UNIT -> COMPILATION_UNIT [4:0]
|--CLASS_DEF -> CLASS_DEF [4:0]
|   |--MODIFIERS -> MODIFIERS [4:0]
|   |   |--BLOCK_COMMENT_BEGIN -> /* [1:0]
|   |   |   |--COMMENT_CONTENT -> *\n * This is 1 version\n  [1:2]
|   |   |   |   `--JAVADOC -> JAVADOC [1:3]
|   |   |   |       |--NEWLINE -> \n [1:3]
|   |   |   |       |--LEADING_ASTERISK ->  * [2:0]
|   |   |   |       |--TEXT ->  This is 1 version [2:2]
|   |   |   |       |--NEWLINE -> \n [2:20]
|   |   |   |       |--TEXT ->   [3:0]
|   |   |   |       `--EOF -> <EOF> [3:1]
|   |   |   `--BLOCK_COMMENT_END -> */ [3:1]
|   |   `--LITERAL_PUBLIC -> public [4:0]
|   |--LITERAL_CLASS -> class [4:7]
|   |--IDENT -> Test [4:13]
|   `--OBJBLOCK -> OBJBLOCK [4:18]
|       |--LCURLY -> { [4:18]
|       `--RCURLY -> } [5:0]
`--CLASS_DEF -> CLASS_DEF [10:0]
    |--MODIFIERS -> MODIFIERS [10:0]
    |--BLOCK_COMMENT_BEGIN -> /* [7:0]
    |   |--COMMENT_CONTENT -> *\n * This is 1.0 version\n  [7:2]
    |   |   `--JAVADOC -> JAVADOC [7:3]
    |   |       |--NEWLINE -> \n [7:3]
    |   |       |--LEADING_ASTERISK ->  * [8:0]
    |   |       |--TEXT ->  This is 1.0 version [8:2]
    |   |       |--NEWLINE -> \n [8:22]
    |   |       |--TEXT ->   [9:0]
    |   |       `--EOF -> <EOF> [9:1]
    |   `--BLOCK_COMMENT_END -> */ [9:1]
    |--LITERAL_CLASS -> class [10:0]
    |--IDENT -> Test2 [10:6]
    `--OBJBLOCK -> OBJBLOCK [10:12]
        |--LCURLY -> { [10:12]
        `--RCURLY -> } [11:0]
```
![Screenshot from 2022-03-14 10-15-22](https://user-images.githubusercontent.com/73882557/158106486-b8ddb83a-d1e7-4b7e-8cad-99e295220526.png)

Summary is only displayed till period, eg-
```
/**
 * This is 1 version. This is after the summary, won't be a part of it.
 */
public class Test {
}
```
![Screenshot from 2022-03-14 10-17-03](https://user-images.githubusercontent.com/73882557/158106654-0b8751f0-beaa-4a08-a0e4-8b8008531177.png)

```
/**
 * This is 1.0 version
 */
```
The complete sentence is displayed and not till 1 but checkstyle consider `.` after 1 as period and gives a false negative.

**Comment by romani** [comment]
We need to be aligned with javadoc tool.

**Comment by Kevin222004** [comment]
@Vyom-Yadav https://github.com/checkstyle/checkstyle/issues/11407#issuecomment-1066028995 if you are not continuing, should I work on this

**Comment by Vyom-Yadav** [comment]
Yes, go ahead, make sure you support other locales too (https://github.com/checkstyle/checkstyle/pull/11383#discussion_r825385669)

**Comment by Kevin222004** [comment]
Yes, I will study https://github.com/checkstyle/checkstyle/pull/11383#discussion_r825385669 properly

**Comment by Kevin222004** [comment]
@Vyom-Yadav what should be expected in this case 
```
    /**
     * Generates <b>suppressions.xml</b> file, based on violations occurred.
     * See issue
     * <a href="https://github.com/checkstyle/checkstyle/issues/102">#102</a>
     */
    class anything {
    }
```
if we want period at the end I have to make it as 
```
    /**
     * Generates <b>suppressions.xml</b> file, based on violations occurred.
     * See issue
     * <a href="https://github.com/checkstyle/checkstyle/issues/102">#102</a>.
     */
    class anything {
    }
```
period after closing of `</a>` it is fine

**Comment by Vyom-Yadav** [comment]
Add period and then generate summary using the javadoc tool, then see if that period works or not, that will help you decide the cases.

**Comment by romani** [comment]
> period after closing of </a> it is fine

it does not matter as it is second sentence, it is not summary.

**Comment by romani** [comment]
```bash
PS D:\Projects\work_checkstyle> cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
<module name="Checker">
        <module name="SummaryJavadoc"/>
    </module>
</module>
PS D:\Projects\work_checkstyle> cat example.java
/**
 * This is 1 version
 */
public class Test {
}

/**
 * This is 1.0 version
 */
class Test2 {
}
PS D:\Projects\work_checkstyle> java -jar checkstyle-10.21.4-all.jar -c /google_checks.xml MyClass.java
Files to process must be specified, found 0.
PS D:\Projects\work_checkstyle> java -jar .\checkstyle-10.21.4-all.jar -c .\Config.xml .\Example.java
Starting audit...
[ERROR] D:\Projects\work_checkstyle\.\Example.java:1: First sentence of Javadoc is missing an ending period. [SummaryJavadoc]
[ERROR] D:\Projects\work_checkstyle\.\Example.java:7: First sentence of Javadoc is missing an ending period. [SummaryJavadoc]
Audit done.
Checkstyle ends with 2 errors.
```

**Comment by romani** [comment]
```
/var/tmp$ java -jar checkstyle-10.21.1-all.jar -c config.xml Test.java 
Starting audit...
Audit done.

/var/tmp$ java -jar checkstyle-10.21.1-all.jar -c config.xml Test.java 
Starting audit...
[ERROR] /var/tmp/Test.java:3: First sentence of Javadoc is missing an ending period. [SummaryJavadoc]
Audit done.
Checkstyle ends with 1 errors.
roman@p5510:/var/tmp$ cat Test.java 
public class Test {

    /**
     * Generates <b>suppressions.xml</b> file, based on violations occurred
     * See issue
     * <a href="https://github.com/checkstyle/checkstyle/issues/102">#102</a>
     */

 public void foo(){}    
}

/var/tmp$ cat config.xml 
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <module name="TreeWalker">
        <module name="SummaryJavadoc"/>
    </module>
</module>

```

**Comment by romani** [comment]
with dot:
```
~/tmp$ javadoc Test.java
Loading source file Test.java...
Constructing Javadoc information...
Standard Doclet version 11.0.2
Building tree for all the packages and classes...
Generating ./Test.html...
Generating ./package-summary.html...
Generating ./package-tree.html...
Generating ./constant-values.html...
Building index for all the packages and classes...
Generating ./overview-tree.html...
Generating ./index-all.html...
Building index for all classes...
Generating ./allclasses-index.html...
Generating ./allpackages-index.html...
Generating ./deprecated-list.html...
Building index for all classes...
Generating ./allclasses.html...
Generating ./allclasses.html...
Generating ./index.html...
Generating ./help-doc.html...

```

![Image](https://github.com/user-attachments/assets/8bda8616-8ea9-4722-a4d7-e48a41623633)
